coderd/database/dbauthz/crud.go

```diff
@@ -0,0 +1,228 @@
+package dbauthz
+
+import (
+	"cdr.dev/slog"
+	"context"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/rbac"
+)
+
+
+
+// insert runs an rbac.ActionCreate on the rbac object argument before
+// running the insertFunc. The insertFunc is expected to return the object that
+// was inserted.
+func insert[
```

Comment:
I might toss these functions in `dbauthz.go` too... for the reason that `crud.go` doesn't mean much to me if I'm scanning through files, but I'd certainly expect this in `dbauthz.go`.